src/main/scala/rules/QuantifiedChunkSupport.scala

```diff
@@ -1982,19 +1982,25 @@ object quantifiedChunkSupporter extends QuantifiedChunkSupport {
 
     val (receiverTerms, quantVars, cond) = lr match {
       case Left(tuple) => tuple
-      case Right(singletonArguments) =>
+      case Right((singletonArguments, cond)) =>
         return relevantChunks.find { ch =>
           val chunkInfo = ch match {
             case qfc: QuantifiedFieldChunk if qfc.singletonArguments.isDefined =>
-              Some(qfc.singletonArguments.get)
+              Some(qfc.singletonArguments.get, qfc.condition)
             case qpc: QuantifiedPredicateChunk if qpc.singletonArguments.isDefined =>
-              Some(qpc.singletonArguments.get)
+              Some(qpc.singletonArguments.get, qpc.condition)
             case _ => None
           }
           chunkInfo match {
-            case Some(cSingletonArguments) =>
+            case Some((cSingletonArguments, cCond)) =>
+
               val equalityTerm = And(singletonArguments.zip(cSingletonArguments).map { case (a, b) => a === b })
-              val result = v.decider.check(equalityTerm, Verifier.config.checkTimeout())
+              // The conditions of two chunks with the same receivers can differ if they originate from separate branches
```

Comment:
Would it also be possible to join the chunks if the condition is not the same? I.e., add up the permission amounts under their respective conditions, and assume the equality conditionally?

Reply:
Yes, but this would require a bit more changes. Since I’m currently working on a more general solution for merging QP chunks, which will also modify findChunk(), I’d prefer to push this small fix now to address the immediate issue and include the broader changes in a separate PR once they’re ready?

Reply:
Yes, makes perfect sense.